File: fullstack-example-frontend/todoapp/src/app/common/models/todo-list.ts
import { DateTime } from "luxon";
import { isDateEqual } from "../date-utility";

export class TodoList {
    public static defaultListId = '00000000-0000-0000-0000-111111000000';
    public readonly id: string;
    public readonly name: string;
    public readonly description: string;
    public readonly createdAt: DateTime;
    public readonly modifiedAt: DateTime
    public readonly deletedAt: DateTime | null;
    public readonly color: string;

    constructor(
        id: string,
        name: string,
        description: string,
        createdAt: DateTime,
        modifiedAt: DateTime, 
        deletedAt: DateTime | null,
        color: string
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.createdAt = createdAt;
        this.modifiedAt = modifiedAt;
        this.deletedAt = deletedAt;
        this.color = color;
    }

    public equals(otherList: TodoList): boolean {
        return this.id === otherList.id &&
        this.color === otherList.color &&
        this.name === otherList.name &&
        this.description === otherList.description &&
        isDateEqual(this.createdAt, otherList.createdAt) &&
        isDateEqual(this.modifiedAt, otherList.modifiedAt) &&
        isDateEqual(this.deletedAt, otherList.deletedAt);
    }
}

File: fullstack-example-backend/src/Eisenhower.Todo.Domain/Models/TodoList.cs
namespace Eisenhower.Todo.Domain;

// TODO: implement IEquatable
public class TodoList 
{
    public static Guid DefaultListGuid = new Guid("00000000-0000-0000-0000-111111000000");
    public TodoListId TodoListId { get; private set; }    
    public UserId UserId { get; private set; }
    public string Name { get; private set; }
    public string Description { get; private set; }
    public string Color { get; private set; }
    public DateTime CreatedAt { get; private set; }
    public DateTime ModifiedAt { get; private set; }
    public DateTime? DeletedAt { get; private set; }
    public IEnumerable<TodoItemId> TodoItemIds { get; private set; }

    public TodoList(
        TodoListId id,
        UserId userId,
        string name,
        string description,
        string color,
        DateTime createdAt,
        DateTime modifiedAt,
        DateTime? deletedAt,
        IEnumerable<TodoItemId> todoItemIds
    ) {
        if (string.IsNullOrEmpty(name))
            throw new ArgumentNullException(nameof(name));
        // TODO: validate color
        this.TodoListId = id;
        this.UserId = userId;
        this.Name = name;
        this.Description = description;
        this.Color = color;
        this.CreatedAt = createdAt;
        this.ModifiedAt = modifiedAt;
        this.DeletedAt = deletedAt;
        this.TodoItemIds = todoItemIds;
    }
}